feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,108 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findBestSavingsOffer(basePlan, validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.pricingPhases.filter { pricingPhase ->
+            pricingPhase.priceAmountMicros == 0L
+        }.map {
+            Pair(offer, parseBillPeriodToDays(it.billingPeriod))
+        }.maxByOrNull { it.second }
+    }.maxByOrNull { it.second }?.first
+}
+
+@Suppress("ReturnCount")
+private fun findBestSavingsOffer(
+    basePlan: GoogleSubscriptionOption,
+    offers: List<GoogleSubscriptionOption>
+): GoogleSubscriptionOption? {
+    val basePlanPricingPhase = basePlan.pricingPhases.firstOrNull() ?: return null
+
+    val longestOfferTotalDays = offers
+        .mapNotNull { subscriptionOption ->
+            subscriptionOption.pricingPhases.sumOf { it.totalNumberOfDays() ?: 0 }
+        }.maxByOrNull { it } ?: 0
```

Comment:
this might be _really_ annoying but i'm kind of wanting this to be its own separate testable function 😬